plugins/radare2/radare2_main.ml

```diff
@@ -0,0 +1,88 @@
+open Bap_core_theory
+open Core_kernel
+open Bap_future.Std
+open Bap.Std
+include Self()
+
+open KB.Syntax
+
+let agent =
+  KB.Agent.register ~package:"bap.std" "radare2-symbolizer"
+    ~desc:"extracts symbols radare2"
+
+let provide_roots funcs =
+  let promise_property slot =
+    KB.promise slot @@ fun label ->
+    KB.collect Theory.Label.addr label >>| function
+    | None -> None
+    | Some addr ->
+      let addr = Bitvec.to_bigint addr in
+      Option.some_if (Hashtbl.mem funcs addr) true in
+  promise_property Theory.Label.is_valid;
+  promise_property Theory.Label.is_subroutine
+
+let extract_name (json : Yojson.t) = 
+  match json with
+  | Yojson.(`Assoc list) -> 
+    (match List.find list ~f:(fun (key, _) -> String.equal key "name") with
+     | Some (_, v) -> (match v with
+         | Yojson.(`String str) -> Some str
+         | _ -> None)
+     | _ -> None)
+  | _ -> None
+
+let extract_addr (json : Yojson.t) = 
+  match json with
+  | Yojson.(`Assoc list) -> 
+    (match List.find list ~f:(fun (key, _) -> String.equal key "vaddr") with
+     | Some (_, v) -> (match v with
+         | Yojson.(`Int i) -> Z.of_int i |> Some
```

Comment:
`x |> Some` is not a valid OCaml, unlike Haskell, constructors are not functions in OCaml and the only valid way to apply the constructor `Some` to an argument is `Some x`.

The reason why the code works for you is that you, accidentally, using the `ppx_pipebang` syntax extension that on the source level substitutes `x |> f` with `f x`. 

So, if you find `x |> Some` appealing, you can just add `ppx_pipebang` to the `BuildDepends` field<sup>1</sup>. You can also use `Some (Z.of_int i)` or `Z.of_int |> Option.some` 

------
<sup>1)</sup> The OASIS OCamlbuild backend is adding the transitive closure of all dependencies to the built target, which also brings this ppx_pipebang dependency and God knows which other dependencies which might overwrite your code. Internally we do not use OASIS/OCamlbuild but OASIS/Omake which is more strict and includes only the specified dependencies and is 10 to 20 times faster (that's the main reason). So for me this code doesn't compile (well unless you add the ppx_pipebang extension). Basically, it means that we need to add yet another workflow that will check that the code is buildable with omake.